Library/Homebrew/cmd/outdated.rb

```diff
@@ -20,94 +22,182 @@ def outdated_args
       switch :verbose,
              description: "Include detailed version information."
       flag   "--json",
-             description: "Print output in JSON format. Currently the default and only accepted "\
-                          "value for <version> is `v1`. See the docs for examples of using the JSON "\
-                          "output: <https://docs.brew.sh/Querying-Brew>"
+             description: "Print output in JSON format. There are two versions: v1 and v2. " \
+                          "v1 is deprecated and is currently the default if no version is specified. " \
+                          "v2 prints outdated formulae and casks. "
       switch "--fetch-HEAD",
              description: "Fetch the upstream repository to detect if the HEAD installation of the "\
                           "formula is outdated. Otherwise, the repository's HEAD will only be checked for "\
                           "updates when a new stable or development version has been released."
+      switch "--greedy",
+             description: "Print outdated casks with `auto_updates` or `version :latest`."
+      switch "--formula",
+             description: "Treat all arguments as formulae."
+      switch "--cask",
+             description: "Treat all arguments as casks."
       switch :debug
       conflicts "--quiet", "--verbose", "--json"
+      conflicts "--formula", "--cask"
     end
   end
 
   def outdated
     outdated_args.parse
 
-    formulae = if args.resolved_formulae.blank?
-      Formula.installed
-    else
-      args.resolved_formulae
-    end
-    if args.json
-      raise UsageError, "invalid JSON version: #{args.json}" unless ["v1", true].include? args.json
+    case json_version
+    when :v1
+      odeprecated "brew outdated --json=v1", "brew outdated --json=v2"
```

Comment:
```suggestion
      # TODO: enable for next major/minor release
      # odeprecated "brew outdated --json=v1", "brew outdated --json=v2"
```

and similar for the no-argument case.